framework/include/libtorch/utils/LibtorchNeuralNetBase.h

```diff
@@ -0,0 +1,38 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifdef LIBTORCH_ENABLED
+
+#pragma once
+
+#include <torch/torch.h>
+#include "MooseError.h"
+
+namespace Moose
+{
+
+/**
+ * This base class is meant to gather the functions and members common in
+ * every neural network based on Libtorch.
+ */
+class LibtorchNeuralNetBase
+{
+public:
+  // Virtual destructor
+  virtual ~LibtorchNeuralNetBase() {}
```

Comment:
You shouldn't need this because you have another virtual function. I'm assuming you added this virtual for casting?

Reply:
Yeah the NN control will own the objects but only through a parent pointer so for deleting derived classes through parent pointers, we need the virtual destructor. I think.